controllers/federation/predict.go

```diff
@@ -157,6 +157,22 @@ func (r *ReconcileFederation) UpdateFunc(e event.UpdateEvent) bool {
 	return true
 }
 
+func (r *ReconcileFederation) ProposalCreateFunc(e event.CreateEvent) bool {
+
+	p := (e.Object).(*current.Proposal)
+	fed, err := current.AddEndAtAnnotation(r.client, *p)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("failed to get federation %s", p.Spec.Federation))
+		return false
+	}
+	if err := r.client.Update(context.TODO(), fed); err != nil {
```

Comment:
why add this annotation again?

Reply:
This is for cases where there is no webhook.